src/compiler/emitter.ts

```diff
@@ -1257,26 +1257,34 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, ge
 
                     // Children
                     if (children) {
-                        for (let i = 0; i < children.length; i++) {
-                            // Don't emit empty expressions
-                            if (children[i].kind === SyntaxKind.JsxExpression && !((<JsxExpression>children[i]).expression)) {
-                                continue;
+                        const isChildEmittable = (child: JsxChild): boolean => {

```

Comment:
It was me. @nojvek I think you should keep the function, but please move it to a function declaration on the same level as `emitJsxElement`. Otherwise, most engines allocate a new closure record per invocation of `emitJsxElement`.